src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,98 @@
+import React, {
+  useCallback,
+  useContext,
+  useMemo,
+} from 'react';
+import classNames from 'classnames';
+
+import { deleteTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { AppContext } from '../AppContext';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoInfo: React.FC<Props> = ({ todo }) => {
+  const {
+    id,
+    title,
+    completed,
+  } = todo;
+
+  const {
+    allTodos,
+    setAllTodos,
+    showError,
+    setShouldShowError,
+    loadingTodosIds,
+    setLoadingTodosIds,
+  } = useContext(AppContext);
+
+  const isLoadingState = useMemo(() => (
+    loadingTodosIds.includes(id)
+  ), [loadingTodosIds]);
+
+  const deleteTodoFromServer = useCallback(async () => {
+    try {
+      setShouldShowError(false);
+      setLoadingTodosIds(prevIds => [...prevIds, id]);
+
+      await deleteTodo(id);
+
+      setAllTodos(prevTodos => prevTodos.filter(prevTodo => (
+        prevTodo.id !== id
+      )));
+    } catch {
+      showError('Unable to delete a todo');
+    } finally {
+      setLoadingTodosIds([0]);
+    }
+  }, [loadingTodosIds, allTodos]);
```

Comment:
The useCallback dependencies array is missing some dependencies. It should include `id`, `showError`, `setShouldShowError`, `setLoadingTodosIds`, and `setAllTodos`. 🧐